server/src/transport/ws.rs

```diff
@@ -308,6 +260,20 @@ pub(crate) async fn background_task<L: Logger>(
 	let result = loop {
 		data.clear();
 
+		let sink_permit_fut = sink.reserve();
+
+		tokio::pin!(sink_permit_fut);
+
+		// Wait until there is a slot in the bounded channel which means that
+		// the underlying TCP socket won't be read.
+		//
+		// This will force the client to read socket on the other side
+		// otherwise the socket will not be read again.
+		let sink_permit = match method_executors.select_with(Monitored::new(sink_permit_fut, &stop_handle)).await {
```

Comment:
we need to use `select_with` here otherwise the pending futures won't be polled

Reply:
such that if the server have a few pending RPC calls and the "bounded channel buffer is full" then `async fn MethodSink::reserve` won't return until there is a slot in the buffer but the underlying futures/rpc calls must still be polled.